hack/fixup-csv

```diff
@@ -0,0 +1,123 @@
+    doc["metadata"]["annotations"]["containerImage"] = doc["spec"]["install"]["spec"]["deployments"][0]["spec"]["template"]["spec"]["containers"][0]["image"]
+    doc["metadata"]["annotations"]["createdAt"] = datetime.utcnow().isoformat(timespec="seconds")
+    doc["metadata"]["annotations"]["support"] = conf["support"]
+    doc["metadata"]["annotations"]["repository"] = conf["repository"]
+    doc["metadata"]["annotations"]["alm-examples"] = literal(conf["alm_examples"])
+    doc["spec"]["icon"] = [{
+        "base64data": logo_b64.decode("ascii"),
+        "mediatype": logo_type,
+    }]
+    doc["spec"]["displayName"] = conf["display_name"]
+    doc["spec"]["description"] = literal(conf["description"])
+    doc["spec"]["minKubeVersion"] = conf["min_kubernetes_version"]
+    doc["spec"]["keywords"] = conf["keywords"]
+    doc["spec"]["links"] = conf["links"]
+    doc["spec"]["maturity"] = conf["maturity"]
+    doc["spec"]["provider"] = conf["provider"]
+    doc["spec"]["maintainers"] = conf["maintainers"]
+    # Workaround for InvalidWebhookDescription. Explained at top of this file.
+    prefixes = dict(
+        ValidatingAdmissionWebhook="validate",
+        MutatingAdmissionWebhook="mutate",
+        ConversionWebhook="convert",
+    )
+    for webhook in doc["spec"]["webhookdefinitions"]:
+        webhook["generateName"] = prefixes[webhook["type"]] + "." + "webhooks.cert-manager.io"
+
+    # Workaround for OLM configuring all the webhook DNS names with a `-service` suffix.
+    # See:
+    # https://github.com/operator-framework/operator-lifecycle-manager/blob/15790a8a2f07fe65a3dbf5a45a54d35e20f2cce9/pkg/controller/install/webhook.go#L254
+    # https://github.com/operator-framework/api/blob/b51286920978aa99422358a3db74392437eaadf0/pkg/operators/v1alpha1/clusterserviceversion_types.go#L207
+    for deployment in doc["spec"]["install"]["spec"]["deployments"]:
+        # [0]["spec"]["template"]["spec"]["containers"][0]["image"]
+        if deployment["name"] != "cert-manager-webhook":
+            continue
+        for container in deployment["spec"]["template"]["spec"]["containers"]:
+            if container["name"] != "cert-manager":
+                continue
+            # Filter out the arguments we're about to change
+            args = [
+                a for a in container["args"]
+                if not a.startswith((
+                        "--dynamic-serving-dns-names",
+                        "--tls-cert-file",
+                        "--tls-private-key-file",
+                ))
+            ]
+            # Add the arguments back with the modified values
+            args.extend([
+                "--dynamic-serving-dns-names=cert-manager-webhook-service.$(POD_NAMESPACE).svc",
+                "--tls-cert-file=/tmp/k8s-webhook-server/serving-certs/tls.crt",
+                "--tls-private-key-file=/tmp/k8s-webhook-server/serving-certs/tls.key",
+            ])
+            container["args"] = args
```

Comment:
There are two issues addressed here:
 * OpenShift / OLM creates its own webhook certificates and injects them into the webhook config and mounts them into the webhook pod. So here I configure the webhook server with  the paths to the OLM mounted key pair files. 
 * It also overrides the webhook service name with a `-service` suffix, which is why I changed the `--dynamic-serving-dns-names` argument, but actually that argument can probably be removed altogether.

This may fix https://github.com/jetstack/cert-manager/issues/4055.